Deliverable 2

icsaszar opened the pull request:
I'm submitting the deliverable earlier for review, because i'd like to know if anything is missing or needs to be changed so i can do it before the deadline.

A question i have is whether i should have the reports and points (last sentence in the Domain model section) in the domain model, or as business logic in the services. I feel that it would fit better in the services because it is generated/calculated (like a post score in the assignment - i.e. there is no Score class).
The points awarded would be loaded from a configuration file (this would be like loading the user score calculation weights from a file in the assignment).

icsaszar committed:
Starts Design and Analysis document

icsaszar committed:
Adds deployemnt and architecture diagrams to visual paradigm project

icsaszar committed:
Adds diagrams to the design document

icsaszar committed:
Adds PDF version of Analysis and Design document for easier viewing

icsaszar committed:
Changes domain model to include multiplicity and field names in relations

icsaszar committed:
Updates PDF export of A&D document

serban-petrescu reviewed (commented):
## Grading

Student: @icsaszar 

| Aspect                             | Pts | Max |
|------------------------------------|-----|-----|
| Base points                        | 4.0 | 4.0 |
| Domain model                       | 1.5 | 1.5 |
| Conceptual architecture            | 1.5 | 1.5 |
| Package design                     | 1.5 | 1.5 |
| Component and deployment diagram   | 1.0 | 1.5 |
| **Final Grade**                    | 9.5 |     |


Observations:
 - Small typos in the conceptual architecture (presentation, persistence layer). But overall nice job!
 - Also, in your conceptual architecture, you have the controllers at the presentation layer, but as you will have to build a web app, I deduce that either you will use Thymeleaf or you omitted one "layer" (which is the client-side UI). 
 - In the deployment diagram, you could add the artifacts that you will have (e.g. a JAR on the server and some HTML + JS on the UI). Also, include the protocol that you want to use (e.g. HTTP or WebSocket).
 - For completeness, it might be an idea to also depict the database --> Usually it is not located on the same machine as the server.

icsaszar committed:
Improves diagrams
Fixes typos in conceptual architecture
Adds missing layer in conceptual architecture
Adds technology details and database to component diagram

serban-petrescu commented:
Regarding your question from the P.R., I understand that these points / reports are not persisted. Nevertheless you already know that they will exist and have a clear role in your application, so you could think about including them in your domain model (and think about the conceptual relationship with the other entities).

serban-petrescu commented:
## Grading

(after improvements)

| Aspect                             | Pts | Max |
|------------------------------------|-----|-----|
| Base points                        | 4.0 | 4.0 |
| Domain model                       | 1.5 | 1.5 |
| Conceptual architecture            | 1.5 | 1.5 |
| Package design                     | 1.5 | 1.5 |
| Component and deployment diagram   | 1.5 | 1.5 |
| **Final Grade**                    | 10.0 |     |

icsaszar committed:
Adds sequence diagrams

icsaszar committed:
Adds data model diagram

icsaszar committed:
Adds class design and testing strategy
